pyvista/core/dataset.py

```diff
@@ -1655,7 +1655,7 @@ def copy_from(self: Self, mesh: _vtk.vtkDataSet, deep: bool = True) -> None:
                 f'The Input DataSet type {type(mesh)} must be '
                 f'compatible with the one being overwritten {type(self)}',
             )
-        if deep:
+        if deep:  # type: ignore[unreachable]
```

Comment:
This looks like `mypy` is assuming the previous line `if not isinstance(self, type(mesh)):` is always true. Maybe there's a way to modify the check? E.g. maybe `issubclass` instead of `isinstance`?

Reply:
this is ok, we can fix later